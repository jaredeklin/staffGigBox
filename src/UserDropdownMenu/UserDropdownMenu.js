import React, { Component } from 'react';
import PropTypes from 'prop-types';

class UserDropdownMenu extends Component {
  componentDidMount() {
    document.addEventListener('click', this.handleClick, false);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClick, false);
  }

  handleClick = event => {
    if (event.target.parentNode.className === 'dropdown-content') {
      return null;
    } else {
      this.props.handleDropdown();
    }
  };

  render() {
    return (
      <div className="dropdown-content">
        <li onClick={this.props.logout}>Log Out</li>
      </div>
    );
  }
}

export default UserDropdownMenu;

UserDropdownMenu.propTypes = {
  handleDropdown: PropTypes.func,
  logout: PropTypes.func
};
