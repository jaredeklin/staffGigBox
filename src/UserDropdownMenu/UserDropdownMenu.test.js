import React from 'react';
import { shallow } from 'enzyme';
import UserDropdownMenu from './UserDropdownMenu';

describe('UserDropdownMenu', () => {
  let wrapper;
  let mockHandleDropdown = jest.fn();
  let mockLogout = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <UserDropdownMenu
        handleDropdown={mockHandleDropdown}
        logout={mockLogout}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleClick', () => {
    it('should close the dropdown when clicked outside container', () => {
      const mockEvent = { target: { parentNode: { className: 'app' } } };

      wrapper.instance().handleClick(mockEvent);
      expect(mockHandleDropdown).toHaveBeenCalled();
    });

    it('should do nothing when clicked in the dropdown container', () => {
      const mockEvent = {
        target: { parentNode: { className: 'dropdown-content' } }
      };

      expect(wrapper.instance().handleClick(mockEvent)).toEqual(null);
    });
  });

  describe('componentWillUnmount', () => {
    it('should remove the event listener', () => {
      document.removeEventListener = jest.fn();

      wrapper.instance().componentWillUnmount();
      expect(document.removeEventListener).toHaveBeenCalled();
    });
  });
});
