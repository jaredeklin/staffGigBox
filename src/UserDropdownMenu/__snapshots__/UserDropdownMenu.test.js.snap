// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDropdownMenu should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdownMenu
    handleDropdown={[Function]}
    logout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <li
        onClick={[Function]}
>
        Log Out
</li>,
      "className": "dropdown-content",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Log Out",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Log Out",
      "type": "li",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li
          onClick={[Function]}
>
          Log Out
</li>,
        "className": "dropdown-content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log Out",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Log Out",
        "type": "li",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
